Source/EventAnalysis/KTIterativeTrackClustering.cc

```diff
@@ -0,0 +1,374 @@
+            newTrack.SetStartTimeInRunCSigma( oldTrack.GetStartTimeInRunCSigma());
+            newTrack.SetStartFrequencySigma( oldTrack.GetStartFrequencySigma());
+            newTrack.SetSlope( (newTrack.GetEndFrequency() - newTrack.GetStartFrequency())/(newTrack.GetEndTimeInRunC() - newTrack.GetStartTimeInRunC()));
+
+        }
+        if (oldTrack.GetEndTimeInRunC() > newTrack.GetEndTimeInRunC())
+        {
+            newTrack.SetEndTimeInRunC( oldTrack.GetEndTimeInRunC());
+            newTrack.SetEndFrequency( oldTrack.GetEndFrequency());
+            newTrack.SetEndTimeInRunCSigma( oldTrack.GetEndTimeInRunCSigma());
+            newTrack.SetEndFrequencySigma( oldTrack.GetEndFrequencySigma());
+            newTrack.SetSlope( (newTrack.GetEndFrequency() - newTrack.GetStartFrequency())/(newTrack.GetEndTimeInRunC() - newTrack.GetStartTimeInRunC()));
+
+        }
+
+        newTrack.SetTotalPower( newTrack.GetTotalPower() + oldTrack.GetTotalPower());
+        newTrack.SetTimeLength( newTrack.GetEndTimeInRunC() - newTrack.GetStartTimeInRunC());
+        newTrack.SetSlopeSigma( sqrt(newTrack.GetSlopeSigma()*newTrack.GetSlopeSigma() + oldTrack.GetSlopeSigma()*oldTrack.GetSlopeSigma()));
+    }
+
+
+    bool KTIterativeTrackClustering::DoTheyMatch(KTProcessedTrackData& Track1, KTProcessedTrackData& Track2)
+    {
+        bool slope_condition1 = std::abs(Track1.GetEndFrequency()+Track1.GetSlope()*(Track2.GetStartTimeInRunC()-Track1.GetEndTimeInRunC()) - Track2.GetStartFrequency())<fFrequencyAcceptance;
+        bool slope_condition2 = std::abs(Track2.GetStartFrequency()-Track2.GetSlope()*(Track2.GetStartTimeInRunC()-Track1.GetEndTimeInRunC()) - Track1.GetEndFrequency())<fFrequencyAcceptance;
+        bool time_gap_in_line = Track1.GetEndTimeInRunC() <= Track2.GetStartTimeInRunC();
+        bool gap_smaller_than_limit = std::abs(Track2.GetStartTimeInRunC() - Track1.GetEndTimeInRunC())<fTimeGapTolerance;
+
+        if (time_gap_in_line and gap_smaller_than_limit and (slope_condition1 or slope_condition2))
+        {
+            return true;
+        }
+
+        slope_condition1 = std::abs(Track2.GetEndFrequency()+Track2.GetSlope()*(Track1.GetStartTimeInRunC()-Track2.GetEndTimeInRunC()) - Track1.GetStartFrequency())<fFrequencyAcceptance;
+        slope_condition2 = std::abs(Track1.GetStartFrequency()-Track1.GetSlope()*(Track1.GetStartTimeInRunC()-Track2.GetEndTimeInRunC()) - Track2.GetEndFrequency())<fFrequencyAcceptance;
+        time_gap_in_line = Track2.GetEndTimeInRunC() <= Track1.GetStartTimeInRunC();
+        gap_smaller_than_limit = std::abs(Track1.GetStartTimeInRunC() - Track2.GetEndTimeInRunC())<fTimeGapTolerance;
+
+        if (time_gap_in_line and gap_smaller_than_limit and (slope_condition1 or slope_condition2))
+        {
+            return true;
+        }
+
+    return false;
+    }
+
+    bool KTIterativeTrackClustering::DoTheyOverlap(KTProcessedTrackData& Track1, KTProcessedTrackData& Track2)
```

Comment:
Variable names to fix: `Track1`, `Track2`

I haven't looked through the calculations in detail, but if there's anything that can be pre-calculated instead of calculated more than once, that can be worthwhile.

I think this can be a const function with both arguments as const references